modules/xml/src/smithy4s/xml/internals/XmlEncoderSchemaVisitor.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.xml
+package internals
+
+import cats.MonoidK
+import cats.syntax.all._
+import smithy.api.XmlAttribute
+import smithy.api.XmlFlattened
+import smithy.api.XmlName
+import smithy4s.schema._
+import smithy4s.{Schema => _, _}
+
+import XmlDocument._
+import cats.kernel.Monoid
+
+private[smithy4s] object XmlEncoderSchemaVisitor extends XmlEncoderSchemaVisitor
+
+private[smithy4s] abstract class XmlEncoderSchemaVisitor
+    extends SchemaVisitor[XmlEncoder]
```

Comment:
wanna make this cached?

same question for decoders in the other PR

Reply:
Will do in a later PR, the XML artifact won't be published until we have a protocol interpreter using it